pinot-common/src/main/java/org/apache/pinot/common/http/MultiHttpRequest.java

```diff
@@ -109,7 +110,11 @@ public <T extends HttpRequestBase> CompletionService<MultiHttpRequestResponse> e
             .build(); // setting the socket
 
     HttpClientBuilder httpClientBuilder =
-        HttpClients.custom().setConnectionManager(_connectionManager).setDefaultRequestConfig(defaultRequestConfig);
+        HttpClients.custom().setConnectionManager(_connectionManager)
+            // The `setSSLSocketFactory` has to be set after `setConnectionManager` due to the bug described here:
+            // https://github.com/apache/pinot/issues/13431#issuecomment-2178323975
```

Comment:
Same comment as another place regarding the comment.

Reply:
https://github.com/apache/pinot/pull/13477/commits/15108722de6ac77beb407be8bdffc8be549b1d91